chains/chains.go

```diff
@@ -121,6 +121,53 @@ func Predict(ctx context.Context, c Chain, inputValues map[string]any, options .
 	return outputValue, nil
 }
 
+const _defaultApplyMaxNumberWorkers = 5
+
+// Apply executes the chain for each of the inputs asynchronously.
+func Apply(ctx context.Context, c Chain, inputValues []map[string]any, maxWorkers int, options ...ChainCallOption) ([]map[string]any, error) { // nolint:lll
```

Comment:
One for a different PR, but it'd be good to enable timeouts/cancellations using the `ctx` so chains don't run indefinitely (if that ever happens).

Reply:
@tobiade can you check out #178?

Reply:
Thanks @FluffyKebab - left some comments.